### PR TITLE
feat(docs): document policy groups

### LINKED ISSUE #1488: document policy groups in reference documentation
After merging https://github.com/chainloop-dev/chainloop/pull/1465, we should document the whole feature.

### PR DESCRIPTION
**Pull request by jiparis**
Closes #1488

### CODE REVIEW

**Review by javirln — APPROVED** [review]

File: docs/docs/reference/policy-groups.mdx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+apiVersion: workflowcontract.chainloop.dev/v1
+kind: PolicyGroup
+metadata:
+  name: sbom-quality
+  description: This policy group applies a number of SBOM-related policies
+  annotations:
+    category: SBOM
+spec:
+  inputs:
+    - name: bannedlicenses
```
Comment (migmartri) [inline docs/docs/reference/policy-groups.mdx:20]: can you use the `-` in the examples? :) 

```
banned-licenses
```

File: docs/docs/reference/policy-groups.mdx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+---
+title: Policy Groups
```
Comment (migmartri) [inline docs/docs/reference/policy-groups.mdx:2]: I'd make this part of the `policies` reference guide. Or at least make sure that somebody reading the policies part don't miss this one.

**Review by migmartri — APPROVED** [review]

File: docs/docs/reference/policy-groups.mdx (hunk 1 above)
Comment (migmartri) [reply docs/docs/reference/policy-groups.mdx:20]: ok, it seems it's not valid, can we use camelcase? would the validations support that? I remember seeing some dns validation

Comment (jiparis) [reply docs/docs/reference/policy-groups.mdx:20]: Yes, currently it uses dns validation. We'll need to remove it and provide a custom validation instead.

Comment (jiparis) [reply docs/docs/reference/policy-groups.mdx:20]: For now I'll put it in camelCase

**Pull request merged**